### PR TITLE
[LinkPresentation] Add Xcode 11 Beta 1 & Beta 2 bindings

### PR DESCRIPTION
**Pull request by tj-devel709**
Inserted NoWatch since UIView is not available on watch

### CODE REVIEW

File: src/Constants.watch.cs.in [hunk 1]
```diff
@@ -63,6 +63,6 @@ namespace ObjCRuntime {
 
 		// WatchOS 6
 		public const string SoundAnalysisLibrary = "/System/Library/Frameworks/SoundAnalysis.framework/SoundAnalysis";
-
+		public const string LinkPresentationLibrary = "/System/Library/Frameworks/LinkPresentation.framework/LinkPresentation";
```
Comment (spouliot) [inline src/Constants.watch.cs.in]: Headers are **not** available on watchOS (only iOS and macOS)
Sadly you can't trust availability macros over header presence :|

Comment (spouliot) [reply src/Constants.watch.cs.in]: you did not add it to `src/frameworks.sources` do it's one less thing to remove ;-)

File: src/linkpresentation.cs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+using System;
+using CoreGraphics;
+using Foundation;
+using ObjCRuntime;
+
+namespace LinkPresentation {
+
+	[ErrorDomain ("LPErrorDomain")]
+	[NoWatch]
```
Comment (spouliot) [inline src/linkpresentation.cs]: you do not have to remove them - they just won't do anything if not part of the watchOS build (from `src/frameworks.sources`)

File: src/linkpresentation.cs [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+		IntPtr Constructor (CGRect rect);
+
+		[Export ("initWithURL:")]
+		IntPtr Constructor (NSUrl url);
+
+		[Export ("initWithMetadata:")]
+		IntPtr Constructor (LPLinkMetadata metadata);
+
+		[Export ("metadata", ArgumentSemantic.Copy)]
+		LPLinkMetadata Metadata { get; set; }
```
Comment (spouliot) [inline src/linkpresentation.cs]: that's worth a test...

here you have a **non** nullable property but it's not clear what's the default value with the non-`LPLinkMetadata` constructors are used.

File: src/linkpresentation.cs [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+		[NullAllowed, Export ("videoProvider", ArgumentSemantic.Retain)]
+		NSItemProvider VideoProvider { get; set; }
+
+		[NullAllowed, Export ("remoteVideoURL", ArgumentSemantic.Retain)]
+		NSUrl RemoteVideoUrl { get; set; }
+	}
+
+	[NoWatch]
+	[Mac (10,15, onlyOn64: true), iOS (13,0)]
+	[BaseType (typeof (UIView))]
```
Comment (spouliot) [inline src/linkpresentation.cs]: not clear if it makes sense to have a default constructor here...

File: src/linkpresentation.cs [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+	interface LPLinkView {
+
+		[Export ("initWithFrame:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (CGRect rect);
```
Comment (spouliot) [inline src/linkpresentation.cs]: good call (in general) but it's not clear if needed
Does iOS let you control the view UI size ? (it does not always for system views)

File: src/Constants.watch.cs.in (hunk 1 above)
Comment (dalexsoto) [inline src/Constants.watch.cs.in]: `LinkPresentation` is not available on WatchOS

```suggestion
```

File: tools/common/Frameworks.cs [hunk 6]
```diff
@@ -362,7 +364,7 @@ public static Frameworks GetwatchOSFrameworks (Application app)
 				{ "MediaPlayer", "MediaPlayer", 5 },
 
 				{ "SoundAnalysis", "SoundAnalysis", 6 },
-
+				{ "LinkPresentation", "LinkPresentation", 6 },
```
Comment (dalexsoto) [inline tools/common/Frameworks.cs]: `LinkPresentation` is not available on WatchOS

```suggestion
```

Comment (spouliot) [inline tools/common/Frameworks.cs]: remove (headers not present on watchOS)

File: src/linkpresentation.cs (hunk 3 above)
Comment (dalexsoto) [reply src/linkpresentation.cs]: Good call, this would indeed almost rule out having default and frame ctors

File: src/linkpresentation.cs (hunk 2 above)
Comment (tj-devel709) [reply src/linkpresentation.cs]: Oh right good point!

File: src/linkpresentation.cs (hunk 3 above)
Comment (tj-devel709) [reply src/linkpresentation.cs]: So it seems that the default constructors creates a non null LPLinkMetadata, but all of the LPLinkMetadata's properties are null.

Comment (spouliot) [reply src/linkpresentation.cs]: seems legit then :)

File: src/Constants.watch.cs.in [hunk 7]
```diff
@@ -63,6 +63,5 @@ namespace ObjCRuntime {
 
 		// WatchOS 6
 		public const string SoundAnalysisLibrary = "/System/Library/Frameworks/SoundAnalysis.framework/SoundAnalysis";
-
```
Comment (rolfbjarne) [inline src/Constants.watch.cs.in]: Unnecessary whitespace change.

File: tools/common/Frameworks.cs [hunk 8]
```diff
@@ -362,7 +364,6 @@ public static Frameworks GetwatchOSFrameworks (Application app)
 				{ "MediaPlayer", "MediaPlayer", 5 },
 
 				{ "SoundAnalysis", "SoundAnalysis", 6 },
-
```
Comment (rolfbjarne) [inline tools/common/Frameworks.cs]: Unnecessary whitespace change.

File: src/linkpresentation.cs (hunk 5 above)
Comment (tj-devel709) [reply src/linkpresentation.cs]: So if I call initWithFrame: CGRectMake (5,5,5,5)
I can create a LPLinkView with the size 5,5 at dimensions 5,5

Comment (spouliot) [reply src/linkpresentation.cs]: sounds legit too :)

**Comment by dalexsoto** [comment]
build

**Comment by dalexsoto** [comment]
Needs update to beta 3

**Comment by spouliot** [comment]
@dalexsoto there's no header diff in beta 3 for this framework
